client/client-lib/src/lib.rs

```diff
@@ -104,6 +107,23 @@ pub struct GatewayClientConfig {
     pub node_pub_key: bitcoin::secp256k1::PublicKey,
 }
 
+#[async_trait]
+pub trait IModuleClient {
+    type Module: FederationModule;
```

Comment:
The client traits are landed in master already in `core/api/src/client.rs` . Better just modify them from start?

Also - `I*` must be object safe, so can't use associated types. Client module interface should probably only be tied to the server module via `module_key`. Internally the implementation can reference to shared types or even server-side type via "common" module, but on the interface to the generic code `IModuleClient` this should be invisible.

Reply:
Unless this is just temporary and you plan to land it soon as a incremental step. Then ignore my previous comment.

Reply:
Oh, I wasn't aware we already had the client skeleton in already. I don't plan to mess with the `dyn` version for now, so thx for the info that the `I*` traits are only for the object safe versions. My trait was essentially my first step towards `ClientModulePlugin`.